lib/fs.js

```diff
@@ -2051,16 +2055,16 @@ function writeAll(fd, isUserFd, buffer, offset, length, signal, callback) {
     }
     return;
   }
-  // write(fd, buffer, offset, length, position, callback)
+
+  if (isCustomIterable(buffer)) {
+    writeAllCustomIterable(
+      fd, isUserFd, buffer, offset, length, signal, encoding, callback)
+      .catch((reason) => { throw reason; });
```

Comment:
This might leave the file open, and not call the callback correctly. I think that it makes more sense to call `handleWriteAllErrorCallback` with the error, instead of creating an unhandled error.